MarcOrfilaCarreras/spanish-news missing on HF

### Describe the bug

The dataset is no longer available on huggingface:

```
FAILED tests/test_tasks/test_dataset_on_hf.py::test_dataset_on_hf[dataset_revision419] - AssertionError: Dataset MarcOrfilaCarreras/spanish-news - 0086c197b914690a9dace258a19398890a05299a not available
assert False
```

This issue does not affect any existing benchmarks. 

### To reproduce

Found in commit CI failure in the last few days:
1. https://github.com/embeddings-benchmark/mteb/actions/runs/28749630498/job/85246368987
2. https://github.com/embeddings-benchmark/mteb/actions/runs/29020373561/job/86125362208

### Additional information

SpanishNewsClassification is not usable. However, there is SpanishNewsClassification.v2 which is based from the same dataset, but not marked as superseded. 

Suggested fix: mark as superseded, and maybe skip checking dataset on hf for superseded tasks. 

### Are you interested to contribute a fix for this bug?

Yes

--- a/mteb/tasks/classification/spa/spanish_news_classification.py
+++ b/mteb/tasks/classification/spa/spanish_news_classification.py
@@ -24,7 +24,7 @@ class SpanishNewsClassification(AbsTaskClassification):
         annotations_creators="derived",
         dialect=[],
         sample_creation="found",
-        bibtex_citation=None,
+        bibtex_citation="",
         superseded_by="SpanishNewsClassification.v2",
     )
 

--- a/mteb/tasks/classification/spa/spanish_news_classification.py
+++ b/mteb/tasks/classification/spa/spanish_news_classification.py
@@ -6,10 +6,10 @@ class SpanishNewsClassification(AbsTaskClassification):
     metadata = TaskMetadata(
         name="SpanishNewsClassification",
         description="A Spanish dataset for news classification. The dataset includes articles from reputable Spanish news sources spanning 12 different categories.",
-        reference="https://huggingface.co/datasets/MarcOrfilaCarreras/spanish-news",
+        reference="https://huggingface.co/datasets/mteb/SpanishNewsClassification",
         dataset={
-            "path": "MarcOrfilaCarreras/spanish-news",
-            "revision": "0086c197b914690a9dace258a19398890a05299a",
+            "path": "mteb/SpanishNewsClassification",
+            "revision": "9f568e396857395bd803c63452b83316124d31c9",
         },
         type="Classification",
         category="t2c",
@@ -24,19 +24,9 @@ class SpanishNewsClassification(AbsTaskClassification):
         annotations_creators="derived",
         dialect=[],
         sample_creation="found",
-        bibtex_citation="""
-        """,
+        bibtex_citation=None,
         superseded_by="SpanishNewsClassification.v2",
     )
-
-    def dataset_transform(
-        self,
-        num_proc: int | None = None,
-    ):
-        self.dataset = self.dataset.rename_columns({"category": "label"})
-        self.dataset = self.stratified_subsampling(
-            self.dataset, seed=self.seed, splits=["train"]
-        )
 
 
 class SpanishNewsClassificationV2(AbsTaskClassification):

